File: src/main/java/com/nndmove/app/service/dto/PasswordChangeDTO.java
package com.nndmove.app.service.dto;

import java.io.Serializable;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO representing a password change required data - current and new password.
 */
@Getter
@Setter
@NoArgsConstructor
public class PasswordChangeDTO implements Serializable {

    private static final long serialVersionUID = 1L;

    private String currentPassword;
    private String newPassword;

    public PasswordChangeDTO(String currentPassword, String newPassword) {
        this.currentPassword = currentPassword;
        this.newPassword = newPassword;
    }
}

File: src/main/java/com/nndmove/app/service/dto/PasswordChangeDTO.kt
package com.nndmove.app.service.dto

import java.io.Serializable

/**
 * A DTO representing a password change required data - current and new password.
 */
class PasswordChangeDTO : Serializable {
    @JvmField
    var currentPassword: String? = null
    @JvmField
    var newPassword: String? = null

    constructor(currentPassword: String?, newPassword: String?) {
        this.currentPassword = currentPassword
        this.newPassword = newPassword
    }

    constructor()

    companion object {
        private const val serialVersionUID = 1L
    }
}
